packages/app-layout/src/vaadin-app-layout.js

```diff
@@ -548,6 +568,58 @@ class AppLayout extends ElementMixin(ThemableMixin(mixinBehaviors([IronResizable
     // TODO(jouni): ARIA attributes. The drawer should act similar to a modal dialog when in ”overlay” mode
   }
 
+  /**
+   * Returns a promise that resolves when the drawer opening/closing CSS transition ends.
+   *
+   * The method relies on the `--vaadin-app-layout-transition` CSS variable to detect whether
+   * the drawer has a CSS transition that needs to be awaited. If the CSS variable equals `none`,
+   * the promise resolves immediately.
+   *
+   * @return {Promise}
+   * @private
+   */
+  __drawerTransitionComplete() {
+    return new Promise((resolve) => {
+      if (getComputedStyle(this).getPropertyValue('--vaadin-app-layout-transition').trim() === 'none') {
```

Comment:
Why don't we also use `_getCustomPropertyValue` helper here?

```suggestion
      if (this._getCustomPropertyValue('--vaadin-app-layout-transition') === 'none') {
```

Reply:
Good point, I didn't know that there is such a one.

Reply:
Fixed.